hms-app/src/components/RelevantIdeasLoader.tsx

```diff
@@ -39,52 +39,44 @@ export default function RelevantIdeasLoader({
     requiredSkills: [],
     category: undefined,
     creationDate: new Date(),
-  } as Idea)
+  })
 
-  const loadSelectedHackathonData = () => {
-    getHackathonDetails(selectedHackweek).then((data) => {
-      setHackathonData({
-        id: data.id,
-        title: data.title,
-        startDate: data.startDate,
-        endDate: data.endDate,
-        participants: data.participants,
-        categories: data.categories,
-        ideas: data.ideas,
-      })
+  const loadSelectedHackathon = () => {
+    getHackathonDetails(selectedHackathonId).then((data) => {
+      setHackathonData(data)
     })
   }
 
   const loadRelevantIdeaDetails = () => {
     hackathonData.ideas?.map((ideaPreviews) => {
       getIdeaDetails(ideaPreviews.id).then((data) => {
-        setIdeaData({
-          id: data.id,
-          owner: data.owner,
-          hackathon: data.hackathon,
-          participants: data.participants,
-          title: data.title,
-          description: data.description,
-          problem: data.problem,
-          goal: data.goal,
-          requiredSkills: data.requiredSkills,
-          category: data.category,
-          creationDate: data.creationDate,
-        })
-        setIsLoading(false)
+        setIdeaData(data)
+        setIsThisLoading(false)
       })
     })
   }
 
   useEffect(() => {
-    loadSelectedHackathonData()
-  }, [selectedHackweek])
+    localStorage.getItem(selectedHackathonId)
+      ? setHackathonData(JSON.parse(localStorage.getItem(selectedHackathonId)!))
+      : loadSelectedHackathon()
+  }, [selectedHackathonId])
+
+  useEffect(() => {
+    localStorage.getItem(hackathonData.id)
+      ? console.log(
+          'id exist',
+          JSON.parse(localStorage.getItem(hackathonData.id)!)
+        )
+      : localStorage.setItem(hackathonData.id, JSON.stringify(hackathonData))
+  }, [hackathonData])
```

Comment:
so... when there is no value in local storage this will set it, but if there is and hackathonData changes it won't do anything?  Shouldn't we overwrite the value when it changes?